packages/x-charts/src/ChartsXAxis/ChartsXAxis.tsx

```diff
@@ -44,44 +44,54 @@ function addLabelDimension(
   }: Pick<ChartsXAxisProps, 'tickLabelInterval' | 'tickLabelStyle'> &
     Pick<AxisDefaultized, 'reverse'> & { isMounted: boolean },
 ): (TickItemType & LabelExtraData)[] {
-  const withDimension = xTicks.map((tick) => {
+  const getTickSize = (tick: TickItemType) => {
     if (!isMounted || tick.formattedValue === undefined) {
-      return { ...tick, width: 0, height: 0 };
+      return { width: 0, height: 0 };
     }
+
     const tickSizes = getWordsByLines({ style, needsComputation: true, text: tick.formattedValue });
+
     return {
-      ...tick,
       width: Math.max(...tickSizes.map((size) => size.width)),
       height: Math.max(tickSizes.length * tickSizes[0].height),
     };
-  });
+  };
 
-  if (typeof tickLabelInterval === 'function') {
-    return withDimension.map((item, index) => ({
-      ...item,
-      skipLabel: !tickLabelInterval(item.value, index),
-    }));
-  }
+  // FIXME: Add this back
+  // if (typeof tickLabelInterval === 'function') {
+  //  return withDimension.map((item, index) => ({
+  //    ...item,
+  //    skipLabel: !tickLabelInterval(item.value, index),
+  //  }));
+  // }
```

Comment:
Having this function will cause some issues because we'll need to potentially provide all `width` and `height` values. One thing that comes to mind is that we might create a lazy getter that will only measure the text when accessing the property, but not sure if there might be unwanted consequences.

Reply:
Why would it require to get the dimension of the label?

It takes the value and the index. hen if `tickLabelInterval` returns false you can ignore the tick because label will not be render

Reply:
> Why would it require to get the dimension of the label?

If it doesn't, it would be a breaking change. Current user functions might be relying on `width` and `height` that are spread onto `item` by the `map` in line 47 of the `addLabelDimension` function.

> It takes the value and the index. hen if tickLabelInterval returns false you can ignore the tick because label will not be render

Yeah, but users might be using `width` or `height` to make decisions on whether to skip the label.

Reply:
Another option is to provide a second argument to the `tickLabelInterval` function for users to optionally obtain the label width. 

```ts
return withDimension.map((item, index) => ({
  ...item,
  skipLabel: !tickLabelInterval(item.value, index, { getLabelSize }),
}));
```

Reply:
I thought the same, but after a second look, the first argument is `item.value` and not item. So user don't have access to `item.width`